docs/design-documents/platform/clocks/clocks.md

```diff
@@ -0,0 +1,418 @@
+
+# Usage scenarios and examples
+
+### Accessing Chrono APIs
+
+Using the Chronos API needs a little thought about namespaces. This section covers that; the following sections will assume the complete Chrono API has been imported.
+
+By default C++14 duration literals will be available after including `mbed.h` due to it including files that use `<chrono>` and having a `using namespace std` directive:
+
+```C++
+#include "mbed.h"
+auto delay = 5ms;
+```
+
+If `MBED_NO_GLOBAL_USING_DIRECTIVE` is defined, or you are not including `mbed.h`, you will need your own directive, and possibly own include, to use the literals:
+
+```C++
+#include <chrono> 
+using namespace std::chrono_literals;
+auto delay = 5ms;
+```
+
+All Chrono APIs are in the namespace `std::chrono`, so it can be convenient to have a directive to import this complete namespace to reduce typing:
+
+```C++
+#include <chrono>
+using namespace std::chrono; // provides both types like std::chrono::microseconds and the chrono_literals
+
+microseconds t = 1h;
+printf("1 hour is %d us\n", int(t.count()));
+```
+
+Namespace directives should only be used in source files to avoid unexpected name collision for users. Header files intended for general use will need to fully qualify their use of Chrono with `std::chrono` prefixes. This also precludes use of literals:
+
+```C++
+// In a header file
+#include <chrono>
+void my_function(std::chrono::microseconds timeout = std::chrono::seconds{1});
+```
+
+### Using the RTOS clock
+
+**Basic delays**
+
+```C++
+for (;;) {
+    LED1 = !LED1;
+    ThisThread::sleep_for(500ms);
+}
+```
+
+**Prefer absolute time**
+
+Using absolute time aids precision, guaranteeing that there is no drift due to code execution time. This applies to all clocks.
+
+```C++
+auto next_wake = Kernel::Clock::now();
+for (;;) {
+    LED1 = !LED1;
+    next_wake += 500ms;
+    ThisThread::sleep_until(next_wake);
+}
+```
+
+Periodic callback APIs such as `Ticker` or `EventQueue::call_every` use this mechanism. If manually rescheduling events yourself, you should use the same technique, rather than using `call_in` within a callback.
+
+**Measuring elapsed time via the RTOS**
+
+```C++
+auto start = Kernel::Clock::now(); // type will be Kernel::Clock::time_point
+do_operation();
+milliseconds elapsed_time = Kernel::Clock::now() - start;
+printf("elapsed time = %d ms\n", int(elapsed_time.count()));
+```
+
+Explicitly stating that `elapsed_time` is `milliseconds` provides a cross-check that the `count()` for the print really is milliseconds. If the clock had lower resolution, it would be implicitly converted to milliseconds. If the clock had higher resolution, it would be a compilation error - see "being more generic" below to handle this.
+
+### Using the high-resolution clock
+
+**Measuring elapsed time using Timer**
+```C++
```

Comment:
some example have a line break before the code block, some don't.

when converting to commonmark with pandoc, a new line is added after `...Timer**`.